### PR TITLE
Mock Test Proposed fix

### PR DESCRIPTION
**Pull request by amar-mesic**
In the code example for mock tests, the filter method was incorrectly mocked. While this causes no failures, it is inconsistent and subtracts from the "reality" of testing. I fixed this along with all associated text. This will also give students the opportunity to verify that no method was called.

Additionally, some typo fixes.

This PR can be revised if the fix can be improved.

### CODE REVIEW

**Comment by amar-mesic** [comment]
Found additional typo in Java Code: 'List<'

File: chapters/pragmatic-testing/test-doubles.md [hunk 1]
```diff
@@ -243,13 +243,12 @@ public class SAPInvoiceSenderTest {
     final var steve = new Invoice("Steve", 99);
     final var arie = new Invoice("Arie", 300);
```
Comment (mauricioaniche) [inline chapters/pragmatic-testing/test-doubles.md:244]: We can remove this line!

File: chapters/pragmatic-testing/test-doubles.md [hunk 2]
```diff
@@ -261,14 +260,14 @@ This example illustrates the main difference between stubbing and mocking. Stubb
 Mockito actually enables us to define even more specific expectations. For example, see the expectations below:
 
 ```java
-verify(sap, times(3)).send(any(Invoice.class));
+verify(sap, times(2)).send(any(Invoice.class));
 verify(sap, times(1)).send(mauricio);
 verify(sap, times(1)).send(steve);
-verify(sap, times(1)).send(arie);
+verify(sap, never()).send(arie);
```
Comment (mauricioaniche) [inline chapters/pragmatic-testing/test-doubles.md:266]: if we remove the previous line, we also should remove this one, as there's no `arie` variable anymore

File: chapters/pragmatic-testing/test-doubles.md (hunk 1 above)
Comment (amar-mesic) [reply chapters/pragmatic-testing/test-doubles.md:244]: Line 244 I presume?

**Pull request merged**